File: osserver/message.py

class Message:
    def __init__(self, connection, json):
        self.connection = connection

        self.data     = json.get('data', None)
        self.uuid     = json.get('uuid', None)
        self.provider = json.get('provider', None)
    async def answer (self, data: any):
        await self.connection.answer(self.uuid, data)

File: webos/api/message.ts

export type MessageType = {
    data     : any;
    provider : string,
    answer_to?: number,
    uuid     ?: number
}

export class Message {
    data: any;

    provider : string;
    answer_to: number | undefined;

    uuid: number | undefined;

    constructor (json: MessageType) {
        this.data = json.data;

        this.provider  = json['provider'];
        this.answer_to = json['answer_to'];

        this.uuid = json.uuid;
    }

    answer () {

    }
}
